openmdao/visualization/n2_viewer/tests/test_n2_viewer.py

```diff
@@ -0,0 +1,106 @@
+"""Unit Tests for n2_viewer"""
+import unittest
+import os
+
+import numpy as np
+
+import openmdao.api as om
+from openmdao.test_suite.components.paraboloid import Paraboloid
+from openmdao.visualization.n2_viewer.n2_viewer import _get_viewer_data
+from openmdao.utils.testing_utils import use_tempdirs
+
+@use_tempdirs
+class TestN2Viewer(unittest.TestCase):
+
+    def setUp(self):
+
+        self.filename = "cases.sql"
+        self.recorder = om.SqliteRecorder(self.filename)
+
+    def test_driver_case(self):
+        prob = om.Problem()
+        model = prob.model
+
+        model.add_subsystem('p1', om.IndepVarComp('x', 0.0), promotes=['x'])
+        model.add_subsystem('p2', om.IndepVarComp('y', 0.0), promotes=['y'])
+        model.add_subsystem('comp', Paraboloid(), promotes=['x', 'y', 'f_xy'])
+
+        model.add_design_var('x', lower=0.0, upper=1.0)
+        model.add_design_var('y', lower=0.0, upper=1.0)
+        model.add_objective('f_xy')
+
+        prob.driver = om.DOEDriver(om.PlackettBurmanGenerator())
+        prob.driver.add_recorder(self.recorder)
+
+        prob.setup()
+        prob.run_driver()
+        prob.cleanup()
+
+        data_dict = _get_viewer_data(self.filename, case_id='rank0:DOEDriver_PlackettBurman|3')
```

Comment:
I think your test model is too simple...  I tried the following:
```
        prob = SellarProblem(SellarDerivativesGrouped)

        prob.driver = om.ScipyOptimizeDriver(tol=1e-9, disp=False)
        prob.driver.add_recorder(self.recorder)

        prob.setup()
        prob.run_driver()
        prob.cleanup()

        prob.model.list_outputs()

        cr = om.CaseReader(self.filename)
        first_case = cr.list_cases()[0]

        data_dict = _get_viewer_data(self.filename, case_id=first_case)
```
and got the following error:
```
    data_dict = _get_viewer_data(self.filename, case_id=first_case)
  File "/home/swryan/dev/OpenMDAO/openmdao/visualization/n2_viewer/n2_viewer.py", line 403, in _get_viewer_data
    j['value'] = sys_cases.outputs[j['name']]
  File "/home/swryan/dev/OpenMDAO/openmdao/recorders/case.py", line 962, in __getitem__
    raise KeyError('Variable name "%s" not found.' % key)
KeyError: 'Variable name "v0" not found.'
```